### PR TITLE
Dartdoc bot sanity flexibility

### PR DESCRIPTION
**Pull request by jcollins-g**
This will make running the dartdoc flutter sanity check pass before and after 0.10.  @devoncarew

### CODE REVIEW

**Comment by jcollins-g** [comment]
After some initial flutter newbie workflow mistakes, this is ready now.

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -111,15 +111,32 @@ void createFooter(String footerPath) {
...
+    '$kDocRoot/api/dart-ui/Canvas-class.html',
+    '$kDocRoot/api/dart-ui/Canvas/drawRect.html',
+  ];
+  final List<String> canaries = <String>[
     '$kDocRoot/api/flutter_test/WidgetTester/pumpWidget.html',
     '$kDocRoot/api/material/Material-class.html',
     '$kDocRoot/api/material/Tooltip-class.html',
     '$kDocRoot/api/widgets/Widget-class.html',
   ];
+  bool oldMissing = false;
```
Comment (devoncarew) [inline dev/tools/dartdoc.dart]: could also do: `bool oldMissing = oldSdkCanaries.any((String canary) => !new File(canary).existsSync());`

**Comment by devoncarew** [comment]
lgtm

**Comment by Hixie** [comment]
Any reason not to just roll to the new dartdoc and forget the old one?

**Comment by jcollins-g** [comment]
New one has broken links in many places; sadly it just isn't ready yet.  Some flutter changes (#9468) also broke the new dartdoc harder because canonicalization means that dartdoc just can't lose track of the SDK and still function like it used to.  So another wave of fixes is in progress, and this will help me test it more easily.

**Comment by Hixie** [comment]
Ah, cool. LGTM.

File: dev/tools/dartdoc.dart (hunk 1 above)
Comment (jcollins-g) [inline dev/tools/dartdoc.dart]: skipping on this optimization to keep it clear and simple.

**Comment by jcollins-g** [comment]
I do not have write access to this repository (probably a good idea since I'm still learning flutter's workflow).  So I can not merge this by myself.

**Pull request merged**